agents/pfeiffer_turbo_controller/Pfeiffer_Turbo_Controller_Driver.py

```diff
@@ -0,0 +1,193 @@
+        """
+        
+        PVP._send_data_request(self.session, self.turbo_address, 346)
+        addr, rw, param_num, motor_temp = PVP._read_gauge_response(self.session)
+        
+        return int(motor_temp)
+    
+    
+    def get_turbo_actual_rotation_speed(self):
+        """Gets the current rotation speed of the turbo from the turbo controller.
+        
+        parameters: 
+            None
+            
+        returns:
+            (int) actual_rotation_speed: The current rotation speed of the turbo in Hz.
+        """
+        
+        PVP._send_data_request(self.session, self.turbo_address, 309)
+        
+        addr, rw, param_num, actual_rotation_speed = PVP._read_gauge_response(self.session)
+        
+        return int(actual_rotation_speed)
+    
+    
+    def get_turbo_set_rotation_speed(self):
+        """Gets the the rotation speed that the turbo is set to from the turbo controller.
+            This is the speed in Hz that the turbo motor will spin up to if turned on.
+        
+        parameters: 
+            None
+            
+        returns:
+            (int) set_rotation_speed: The rotation speed that the turbo is set to in Hz
+        """
+        
+        PVP._send_data_request(self.session, self.turbo_address, 308)
+        
+        addr, rw, param_num, set_rotation_speed = PVP._read_gauge_response(self.session)
+        
+        return int(set_rotation_speed)
+    
+    
+    def get_turbo_error_code(self):
+        """Gets the current error code of the turbo from the turbo controller.
+        
+        parameters: 
+            None
+            
+        returns:
+            (int) error_code: the current error code of the turbo.
+        """
+        PVP._send_data_request(self.session, self.turbo_address, 303)
+        
+        
+        addr, rw, param_num, error_code = PVP._read_gauge_response(self.session)
+        
+        return int(error_code)
+    
+    def unready_turbo(self):
+        """Unreadies the turbo. Does not cause the turbo to spin up.
+        
+        parameters: 
+            None
+            
+        returns:
+            (bool) turbo_response: True for successful, False for failure.
+        """
+        
+        PVP._send_control_command(self.session, self.turbo_address, 10, "000000")
+        
+        addr, rw, param_num, turbo_response = PVP._read_gauge_response(self.session)
+        
+        return turbo_response
+    
+    
+    def ready_turbo(self):
+        """Readies the turbo for spinning. Does not cause the turbo to spin up.
+        
+        parameters: 
+            None
+            
+        returns:
+            (bool) turbo_response: True for successful, False for failure.
+        """
+        
+        PVP._send_control_command(self.session, self.turbo_address, 10, "111111")
+        
+        addr, rw, param_num, turbo_response = PVP._read_gauge_response(self.session)
+        
+        return turbo_response
```

Comment:
I think this, and some of the other methods defined to command the turbo, could benefit from checking the returned values from `_read_gauge_response`, similar to what's done in the `Pfeiffer_Vacuum_Protocol.py` file. What happens if the turbo can't be readied? Should we raise an error? What happens in the scenario the turbo can't be readied, but then we call `turn_turbo_motor_on()`?

What brought me to this suggestion was considering the `turn_turbo_on` Task in the Agent, which boils down to:
```
self.turbo.ready_turbo()
time.sleep(1)
self.turbo.turn_turbo_motor_on()
```

What happens if line 1 fails and returns False? We would move right along to trying to turn on the turbo, and then report it turned on in either case. Raising an error here if the turbo couldn't be readied, or couldn't be turned on (in the case of `turn_turbo_motor_on`) would allow us to check for that error and react accordingly in the Agent.

(Structurally I also wonder if one would ever call `turn_turbo_motor_on` without first calling `ready_turbo`, in which case I would personally probably make `turn_turbo_motor_on` call `ready_turbo` and `time.sleep(1)` internally, rather than requiring one do so externally. In either case, checking we were successful in turning on the turbo is important, when that's what we expect the method to do, either here or in the Agent.)

Reply:
I put return statements if either 'ready' or 'on' fail to return with errors.